source/software/matlab_parallel_computing.rst

```diff
@@ -7,119 +7,99 @@ General
-   >> c = parcluster('local');
+    $ # Access
+    $ c.JobStorageLocation = '</path/to/your/custom/jobdir>'
 
-You can set a range of additional properties on each profile. There are two properties that are required, namely ``AccountName`` and ``WallTime``. Set any 
-other to your preference. Once these are saved, they will be kept in the settings of your cluster profile, but they can always be adapted again later.
+Both after changing any of the additional properties or the standard parameters, you will have to save that profile to keep the changes after closing your
+session:
 
 ::
 
-   >> % View the additional properties
-   >> c.AdditionalProperties
-   >> % set the required properties
-   >> c.AdditionalProperties.AccountName = <account-name>;
-   >> c.AdditionalProperties.WallTime = '05:00:00';
-   >> % save the profile
-   >> c.saveProfile
-   
-Now the profile is ready, and you should be able to submit multi-node jobs with Matlab now.
+    $ c.saveProfile;
+
 
 Submitting jobs
 +++++++++++++++
 
 Submitting jobs will happen from within Matlab. There are two ways to do this. You can start a Matlab session on a login node, or you can first request an
 interactive job, where you launch your Matlab session. Which one you choose will depend on the workload you plan to have on the node before submitting a job.
-As for any other work on the cluster, you can always do minor work on the login node, but for heavier calculations (before submitting your multinode job), you should
-use an interactive job. 
+As for any other work on the cluster, you can always do minor work on the login node, but for heavier calculations (before submitting your effective job), you should
+use an interactive job.
 
 In general, we recommend to carry out the examples below via Matlab's command line interface
 (as shown in the configuration section). It is also possible to use the Matlab GUI, in which case it
 is best to do this via a NoMachine connection. If you then need to launch the Matlab GUI in an
-interactive job, don't forget to add ``-X`` option to qsub and keep in mind that the GUI will
-necessarily be less responsive.
+interactive job, don't forget to add ``-X`` option to your `srun` or `salloc` command and keep in mind
+that the GUI will necessarily be less responsive.
 
-Now you can start both interactive and independent batch jobs with the previously configured cluster profile. Follow the steps below to submit a job.
+Now you can start both interactive and independent batch jobs with the previously configured cluster profile. In both cases, you request a number of threads, either using the `parpool` or `batch`
+function. Under the hood, Matlab launches a Slurm job to request resources of the system. Different from submitting a Slurm job yourself, there is no direct way to specify the number of nodes and cores.
+Matlab just requests a number of tasks, and the system gives whatever is available, meaning that cores could be spread out over a number of nodes.
```

Comment:
the good thing is that the cluster configuration file uses `salloc` with the `-N 1` option, right?
If so, then users are bound to as many cores available on a single node, but they are guaranteed to stay within a node!
is that correct?

Reply:
I think the ``salloc`` statement we found was not related to how Matlab submits the jobs eventually, that was a testing step for the connection I think.

Anyway, it is also possible to request more cores than the max on a node, so I don't think Matlab submits with ``-N 1``

Reply:
"Under the hood, Matlab launches a Slurm job to request resources of the system. Different from submitting a Slurm job yourself, there is no direct way to specify the number of nodes and cores. Matlab just requests a number of tasks, and the system gives whatever is available, meaning that cores could be spread out over a number of nodes."

But if this is done on an interactive node, the "whatever is available" is determined by the nodes and cores specified for that initial interactive slurm job, no? If that is not correct, can you explain why (because I seem to be missing some basic understandig of this then). If it is correct, I still think it would be useful to add something about that (see my example in my previous comment).

Reply:
No, it is not limited by the original job. The logic is like this: 
1. You start an interactive job with whatever amount of resources you want (enough decently start Matlab and then perhaps perform some minor computing, e.g. preparation steps for your Matlab job)
2. You open Matlab
3. You use Matlab to submit a new job to the cluster: You use the parpool or batch function to either send a 'srun/salloc' or a 'sbatch' command to the cluster. Just as in any interactive session, you can submit a job to other nodes from there, the difference here is that you use a Matlab function to assemble your proper slurm command instead of typing it in directly. On top of that, the 'parpool' function will connect to the interactive session and allow you to perform Matlab commands there. The 'batch' function will directly submit certain Matlab commands/scripts to run in the background.